compiler/test_intern_string.cpp

```diff
@@ -0,0 +1,25 @@
+#include "intern_string.h"
+#include <assert.h>
+
+int main(int argc, char const *argv[]) {
+  const char *input_1 = "test";
+  const char *input_2 = "test 2";
+  const char *input_3 = "";
+  const char *input_4;
```

Comment:
As it seems from the tests, you assume that this is initialized to `NULL` but this is not necessarily the case (that is because it has undefined value - it just _happens_ that most compilers (especially in unoptimized modes) initialize it to 0 for better predictability).
You can initialize it explicitly (i.e. ` = NULL;`).